src/ai/backend/gateway/vfolder.py

```diff
@@ -968,7 +1039,7 @@ async def download_single(request: web.Request, params: Any, row: VFolderRow) ->
     except (ValueError, FileNotFoundError):
         raise GenericNotFound('The file is not found.')
     if not file_path.is_file():
-        raise InvalidAPIParameters('The file is not a regular file.')
+        return await download_directory_as_archive(request, file_path)
```

Comment:
I think.. it's better *not* to implicitly change the mode of behavior, unless we have an explicit parameter to order so.
Let's add a new API endpoint for downloading directories.

Reply:
I added `archive: boolean` parameter for downloading directory as an archive.